api/controllers/notifications_controller.py

```diff
@@ -33,34 +50,59 @@ def ack(user_id: int, notification_id: int) -> dict:
 # Admin actions
 @cross_origin()
 @admin_required
-def get_all(_, notification_id: int) -> dict:
+def get_all(_) -> dict:
     """Fetch all notifications"""
-    # TODO: Fetch all notifiations on the system
-    # allow quer string params for pagination and filters
-    return {'status': 'OK', 'notification_id': notification_id}, 200
+    page = request.args.get('p', default=1, type=int)
+    limit = request.args.get('limit', default=10, type=int)
+    if limit > 50 and page > 0:
+        return {'error': 'Limit must be less then 51 and page greater then 0'}
+
+    offset = (page - 1) * limit
+    notif_list = Notifications.select(
+        limit=limit,
+        offset=offset,
+        order_by='created_at DESC',
+    )
+
+    return jsonify(notif_list), 200
 
 @cross_origin()
 @admin_required
 def get_notif_details(_, notification_id: int) -> dict:
     """Get notification information"""
-    # TODO: get notifications data
-    return {'status': 'OK', 'id': notification_id}, 200
+    notif_details = Notifications.select_first(
+        id = notification_id
+    )
+
+    if notif_details is None:
+        return {'error': "Notification doesn't exist"}, 401
+
+    return jsonify(notif_details), 200
 
 @cross_origin()
 @admin_required
 def add_notification(_) -> dict:
     """Add new notification"""
-    # TODO: get data provided via json in the request body and
-    # add new notification
-    return {'status': 'OK'}, 201
+    body = json.loads(request.data)
+
+    try:
+        new_notif = Notifications.insert(**body)
+        return {"Success! ID": new_notif}, 200
+    except Exception as e:
+        return {'error': str(e)}
 
 @cross_origin()
 @admin_required
 def update_notification(_, notification_id: int) -> dict:
     """Update notification info"""
-    # TODO: Get data provided via body json and update the data of
-    # provided notification_id
-    return {'status': 'OK', 'id': notification_id}, 200
+    body = json.loads(request.data)
+
+    try:
+        Notifications.update(conditions={'id': notification_id}, new_values=body)
+        notif_update = Notifications.select_first(id=notification_id)
+        return {"Success!": notif_update}, 200
```

Comment:
I don't see any reason for a tag of the data to be "Success!" it does not provide any value, I suggest:
```suggestion
        return jsonify(notif_update), 200
```